FreePIE.Core.Plugins/vigem/DualShockOutputPlugin.cs

```diff
@@ -0,0 +1,274 @@
+            set => controller.SetButtonState(DualShock4Button.Triangle, value);
+        }
+
+        public bool L2
+        {
+            get => (buttons & DualShock4Button.ShoulderLeft.Value) != 0;
+            set => controller.SetButtonState(DualShock4Button.ShoulderLeft, value);
+        }
+
+        public bool R2
+        {
+            get => (buttons & DualShock4Button.ShoulderRight.Value) != 0;
+            set => controller.SetButtonState(DualShock4Button.ShoulderRight, value);
+        }
+
+        public bool options
+        {
+            get => (buttons & DualShock4Button.Options.Value) != 0;
+            set => controller.SetButtonState(DualShock4Button.Options, value);
+        }
+
+
+        public bool share
+        {
+            get => (buttons & DualShock4Button.Share.Value) != 0;
+            set => controller.SetButtonState(DualShock4Button.Share, value);
+        }
+
+
+        public bool up
+        {
+            get => _DpadFlags.HasFlag(dpadFlags.Up);
+            set => _DpadFlags = value ? _DpadFlags | dpadFlags.Up : _DpadFlags & ~dpadFlags.Up;
+        }
+
+        public bool down
+        {
+            get => _DpadFlags.HasFlag(dpadFlags.Down);
+            set => _DpadFlags = value ? _DpadFlags | dpadFlags.Down : _DpadFlags & ~dpadFlags.Down;
+
+        }
+
+        public bool left
+        {
+            get => _DpadFlags.HasFlag(dpadFlags.Left);
+            set => _DpadFlags = value ? _DpadFlags | dpadFlags.Left : _DpadFlags & ~dpadFlags.Left;
+        }
+
+        public bool right
+        {
+            get => _DpadFlags.HasFlag(dpadFlags.Right);
+            set => _DpadFlags = value ? _DpadFlags | dpadFlags.Right : _DpadFlags & ~dpadFlags.Right;
+        }
+
+        public bool leftThumb
+        {
+            get => (buttons & DualShock4Button.ThumbLeft.Value) != 0;
+            set => controller.SetButtonState(DualShock4Button.ThumbLeft, value);
+        }
+
+        public bool rightThumb
+        {
+            get => (buttons & DualShock4Button.ThumbRight.Value) != 0;
+            set => controller.SetButtonState(DualShock4Button.ThumbRight, value);
+        }
+
+        public bool PS
+        {
+            get => (controller.SpecialButtonState & DualShock4SpecialButton.Ps.Value) != 0;
+            set => controller.SetButtonState(DualShock4SpecialButton.Ps, value);
+        }
+
+        public bool touchPad
+        {
+            get => (controller.SpecialButtonState & DualShock4SpecialButton.Touchpad.Value) != 0;
+            set => controller.SetButtonState(DualShock4SpecialButton.Touchpad, value);
+        }
+
+        /// <summary>
+        /// Acceptable values range 0 - 1
+        /// </summary>
+        public double L1
+        {
+            get => controller.LeftTrigger / 255.0;
+            set
+            {
+                if (isBetween(value, 0, 1))
+                {
+                    var v = (byte)ensureMapRange(value, 0, 1, 0, 255);
```

Comment:
why is left trigger handled differently than right trigger?

Reply:
not sure =)  i fixed it.